examples/examples_test.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+set +e
+
+clean () {
+    jobs -p | xargs pkill -P 
+    wait
+}
+
+fail () {
+    echo "$(tput setaf 1) $1 $(tput sgr 0)"
+    clean
+    exit 1
+}
+
+pass () {
+    echo "$(tput setaf 2) $1 $(tput sgr 0)"
+}
+
+EXAMPLES=(
+    "helloworld"
+    "route_guide"
+    "features/authentication"
+    "features/compression"
+    "features/deadline"
+    "features/encryption/TLS"
+)
+
+declare -A EXPECTED_SERVER_OUTPUT=( 
+    ["helloworld"]="Received: world"
+    ["route_guide"]=""
+    ["features/authentication"]="server starting on port 50051..."
+    ["features/compression"]="UnaryEcho called with message \"compress\""
+    ["features/deadline"]=""
+    ["features/encryption/TLS"]=""
+)
+
+declare -A EXPECTED_CLIENT_OUTPUT=(
+    ["helloworld"]="Greeting: Hello world"
+    ["route_guide"]="location:<latitude:416851321 longitude:-742674555 >"
+    ["features/authentication"]="UnaryEcho:  hello world"
+    ["features/compression"]="UnaryEcho call returned \"compress\", <nil>" 
+    ["features/deadline"]="wanted = DeadlineExceeded, got = DeadlineExceeded" 
+    ["features/encryption/TLS"]="UnaryEcho:  hello world"
+)
+
+for example in ${EXAMPLES[@]}; do
+    echo "$(tput setaf 4) testing: ${example} $(tput sgr 0)"
+
+    # Build server
+    if ! go build -o /dev/null ./examples/${example}/*server/*.go; then
+        fail "failed to build server"
+    else
+        pass "successfully built server"
+    fi
+
+    # Build client
+    if ! go build -o /dev/null ./examples/${example}/*client/*.go; then
+        fail "failed to build client"
+    else
+        pass "successfully built client"
+    fi
+    
+    # Start server
+    SERVER_LOG="$(mktemp)"
+    go run ./examples/$example/*server/*.go &> $SERVER_LOG  &
+
+    CLIENT_LOG="$(mktemp)"
+    if ! go run examples/${example}/*client/*.go &> $CLIENT_LOG; then
```

Comment:
Should we have some kind of timeout on this operation in case the client binary hangs?  I *think* you can do:
```
if ! timeout 20 go run examples/${example}/*client/*.go &> $CLIENT_LOG; then
```